edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -75,7 +73,15 @@ public sealed class DeviceScopeIdentitiesCache : IDeviceScopeIdentitiesCache
             Preconditions.CheckNotNull(encryptedStorage, nameof(encryptedStorage));
             Preconditions.CheckNotNull(serviceIdentityHierarchy, nameof(serviceIdentityHierarchy));
             IDictionary<string, StoredServiceIdentity> cache = await ReadCacheFromStore(encryptedStorage);
-            var deviceScopeIdentitiesCache = new DeviceScopeIdentitiesCache(serviceIdentityHierarchy, serviceProxy, encryptedStorage, cache, refreshRate, refreshDelay);
+
+            // Populate the serviceIdentityHierarchy
+            foreach (KeyValuePair<string, StoredServiceIdentity> kvp in cache)
+            {
+                await kvp.Value.ServiceIdentity.ForEachAsync(serviceIdentity => serviceIdentityHierarchy.AddOrUpdate(serviceIdentity));
+            }
```

Comment:
hmm...I don't see the purpose of moving this out of the ctor and then passing in a bool, couldn't the ctor deduce the same state from the cached dictionary?

Reply:
I moved so that it doesn't use Wait in the ctor instead could use await in create. serviceIdentityHierarchy will always contain edgeHub identity because of a changed made by Vadim to add it when edgeHub  service starts (to solve an issue with  connecting upstream)